### PR TITLE
Create A New Category

### LINKED ISSUE #38: Create a Category
As an admin I would like to be able to create a new category so that I can give authors the ability to better classify their posts.

**Given** an admin is on the Category list page  
**When** they select the `Create Category` button  
**Then** they should be directed to a form in which they can enter a new category name  

**Given** an admin has entered a Category name  
**When** they click the `Save` button  
**Then** a new category should be saved to the database  
**And** the admin should be directed to the Category list page

> **NOTE:** For the time being it is acceptable to treat all users as `admin` users. There is a future story about enforcing user permissions.

### PR DESCRIPTION
**Pull request by IsaacWeiser**
#### Changes Made
1. Added file `addCategory.js` to `client` directory.
1. Modified file `CategoryRepository.cs` to include `public void AddCategory(Category category)` and functionality.
2. Modified file `ICategoryRepository.cs` to include `public void AddCategory(Category category)` signature.
3. Modified file `CategoryController.cs` to include `public IActionResult Post (Category category)` and functionality.
4. Modified file `applicationViews.js` to include `route for new category form` and functionality.
5. Modified file `Category.js` to include `button to create new category` and functionality.

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout iw-createCategory
    ```
6. Open a new Terminal tab (⌘T) and navigate to the server directory.
7. Test app functionality.
    > when viewing list of categories (after selecting categories nav option),
    >At the bottom of the categories list you should see a make new category button
    >Click on this button, and you will be directed to a form to create a new  category
    >On this page, enter in the name for the new category
    >Then click save, you should then be redirected to the list of categories
    > Observe your newly created category

8. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

**Comment by NathanCenter** [comment]
Pulled down the branch and ran a test on it all the functionality works out fine by me!

**Review by NathanCenter — APPROVED** [review]

File: Tabloid/client/src/components/AddCategory.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from "react";
+import { addCategory } from "../modules/CategoryManager";
+import { useHistory } from "react-router-dom";
+
+export const AddCategoryForm = () => {
+  const [category, updateCategory] = useState({ name: "" });
```
Comment (joshdbarton) [inline Tabloid/client/src/components/AddCategory.js:6]: Is there anything in this component preventing you from submitting an unnamed (empty string) category?

**Review by joshdbarton — CHANGES_REQUESTED** [review]

File: Tabloid/client/src/components/AddCategory.js (hunk 1 above)
Comment (IsaacWeiser) [reply Tabloid/client/src/components/AddCategory.js:6]: taken care of @joshdbarton

File: Tabloid/client/src/components/AddCategory.js [hunk 2]
```diff
@@ -12,14 +12,25 @@ export const AddCategoryForm = () => {
 
   const submit = (evt) => {
     evt.preventDefault();
-    addCategory(category).then(() => history.push("/category"));
+
+    const input = document.querySelector("#newCatInp").value;
+    if (
+      !input ||
```
Comment (joshdbarton) [inline Tabloid/client/src/components/AddCategory.js]: I think you can shorten this to `!input || !input.trim()`

**Review by joshdbarton — CHANGES_REQUESTED** [review]

File: Tabloid/client/src/components/AddCategory.js (hunk 2 above)
Comment (IsaacWeiser) [reply Tabloid/client/src/components/AddCategory.js]: shortened! @joshdbarton

**Review by joshdbarton — APPROVED** [review]
